src/cluster.c

```diff
@@ -4177,6 +4217,16 @@ void clusterBeforeSleep(void) {
     if (flags & CLUSTER_TODO_UPDATE_STATE)
         clusterUpdateState();
 
+    /* Notify clients subscribed to moved slot events. To avoid flooding the
+     * clients, we only publish the moved slot message if exactly one slot has
+     * been migrated. In cases like failover, clients will receive -MOVED
```

Comment:
Can we consider sending the MOVED_SLOT_MULTIPLE (-2) message anyways? This would give clients a chance to refresh the cluster topology outside the data serving path while not adding too much overhead.

Reply:
We could. We could also notify clients about changed replicas (no slots moved). Mostly we need to decide if we want to send it in the same channel or a different one. WDYT?